### PR TITLE
Add option for loading yang models from classpath in gNMI simulator - master

### PR DESCRIPTION
**Pull request by PeterSuna**
Implement option to load Yang models from class path with  ```schemaServiceConfig``` parameter in gNMI simulator configuration. Both option for loading models from class path or from folder containing models can be used at the same time. Loading YANG models from class path resolve openconfig models duplicity in test resources.

### CODE REVIEW

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import org.opendaylight.yangtools.yang.binding.YangModuleInfo;
+import org.opendaylight.yangtools.yang.model.api.EffectiveModelContext;
+import org.opendaylight.yangtools.yang.model.repo.api.YangTextSchemaSource;
+import org.opendaylight.yangtools.yang.parser.api.YangParserException;
+import org.opendaylight.yangtools.yang.parser.rfc7950.reactor.RFC7950Reactors;
+import org.opendaylight.yangtools.yang.parser.rfc7950.repo.YangStatementStreamSource;
+import org.opendaylight.yangtools.yang.parser.spi.meta.ReactorException;
+import org.opendaylight.yangtools.yang.parser.stmt.reactor.CrossSourceStatementReactor.BuildAction;
+
+public class EffectiveModelContextBuilder {
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java:22]: add javadoc for class - describe class & summarize usage for this class

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java [hunk 2]
```diff
@@ -43,11 +50,41 @@ public static GnmiSimulatorConfiguration loadGnmiSimulatorConfiguration(final In
...
+                gnmiSimulatorConfiguration = setModelsToSimulatorConfig(mapper, schemaServiceElement,
+                        gnmiSimulatorConfiguration);
+            }
+
         } catch (final JsonProcessingException e) {
             throw new RuntimeException(String.format("Cannot bind Json tree to type: %s",
                 io.lighty.modules.gnmi.simulatordevice.config.GnmiSimulatorConfiguration.class), e);
         }
 
+
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java:64]: remove empty lines

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java [hunk 3]
```diff
@@ -43,11 +50,41 @@ public static GnmiSimulatorConfiguration loadGnmiSimulatorConfiguration(final In
...
+    }
+
+    private static GnmiSimulatorConfiguration setModelsToSimulatorConfig(final ObjectMapper mapper,
+            final JsonNode schemaServiceElement, final GnmiSimulatorConfiguration gnmiSimulatorConfiguration)
+            throws JsonProcessingException {
+        if (!schemaServiceElement.has(TOP_LEVEL_MODELS_ELEMENT_NAME)) {
+            LOG.info("Missing [{}] inside element [{}] in gNMI configuration", TOP_LEVEL_MODELS_ELEMENT_NAME,
+                    SCHEMA_SERVICE_ELEMENT_NAME);
+            return gnmiSimulatorConfiguration;
+        }
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java:75]: move negative branch to the end

**Review by vainyksi — CHANGES_REQUESTED** [review]
please summarize changes in pull-request description

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
+package io.lighty.modules.gnmi.simulatordevice.utils;
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java:9]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+     * @return {@link EffectiveModelContext}
+     * @throws EffectiveModelContextBuilderException if models information was not provided or occur any exception
+     *                                               during construct EffectiveModelContext
+     */
+    public EffectiveModelContext build() throws EffectiveModelContextBuilderException {
+        if (this.yangModulesPath == null && this.yangModulesInfo == null) {
+            throw new EffectiveModelContextBuilderException("Cannot create EffectiveModelContext without"
+                    + "yangModulesPath or yangModulesInfo");
+        }
+        BuildAction buildAction = RFC7950Reactors.defaultReactorBuilder().build().newBuild();
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * Construct {@link EffectiveModelContext} from provided yang models in {@link #addYangModulesInfo(Set)} and {@link
+     * #addYangModulesPath(String)}'.
+     *
+     * @return {@link EffectiveModelContext}
+     * @throws EffectiveModelContextBuilderException if models information was not provided or occur any exception
+     *                                               during construct EffectiveModelContext
+     */
+    public EffectiveModelContext build() throws EffectiveModelContextBuilderException {
+        if (this.yangModulesPath == null && this.yangModulesInfo == null) {
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java (hunk 1 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/EffectiveModelContextBuilder.java:34]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java (hunk 2 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java (hunk 3 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/utils/GnmiSimulatorConfUtils.java]: done

**Review by vainyksi — APPROVED** [review]

**Review by vainyksi — APPROVED** [review]

**Pull request merged**